src/components/Proposals/ProposalDetails/VoteActions.js

```diff
@@ -0,0 +1,251 @@
+  VOTE_STATUS_SCHEDULED,
+} from '../disputable-vote-statuses'
+import { dateFormat, toMs } from '../../../utils/date-utils'
+import { useWallet } from '../../../providers/Wallet'
+
+const VoteActions = React.memo(
+  ({ vote, onVoteYes, onVoteNo, onExecute, onChangeVote }) => {
+    const theme = useTheme()
+    const { account } = useWallet()
+    const disputableStatus = DISPUTABLE_VOTE_STATUSES.get(vote.status)
+    const { snapshotBlock, startDate, hasEnded, voterInfo, orgToken } = vote
+
+    if (disputableStatus !== VOTE_STATUS_SCHEDULED) {
+      return <></>
+    }
+
+    if (!account || !voterInfo) {
+      return (
+        <div
+          css={`
+            border-radius: ${RADIUS}px;
+            background: ${theme.background};
+            padding: ${3.5 * GU}px ${10 * GU}px;
+            text-align: center;
+          `}
+        >
+          <div
+            css={`
+              ${textStyle('body1')};
+            `}
+          >
+            You must enable your account to vote on this proposal
+          </div>
+          <div
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.surfaceContentSecondary};
+              margin-top: ${2 * GU}px;
+            `}
+          >
+            Connect to your Ethereum provider by clicking on the{' '}
+            <strong
+              css={`
+                display: inline-flex;
+                align-items: center;
+                position: relative;
+                top: 7px;
+              `}
+            >
+              <IconConnect /> Connect account
+            </strong>{' '}
+            button on the header. You may be temporarily redirected to a new
+            screen.
+          </div>
+        </div>
+      )
+    }
+
+    if (hasEnded) {
+      return (
+        <>
+          {voterInfo.canExecute && (
+            <>
+              <Button mode="strong" onClick={onExecute} wide>
+                Enact this vote
+              </Button>
+              <Info>
+                The voting period is closed and the vote has passed.{' '}
+                <strong>Anyone</strong> can now enact this vote to execute its
+                action.
+              </Info>
+            </>
+          )}
+        </>
+      )
+    }
+
+    if (
+      !hasEnded &&
+      voterInfo.hasVoted &&
+      parseInt(voterInfo.accountBalance) > 0
+    ) {
+      return (
+        <div>
+          <Button
+            mode="strong"
+            onClick={onChangeVote}
+            wide
+            css={`
+              margin-bottom: ${2 * GU}px;
+            `}
+          >
+            Change my vote
+          </Button>
+          <Info>
+            While the voting period is open, you can{' '}
+            <strong>change your vote</strong> as many times as you wish.
+          </Info>
+        </div>
+      )
+    }
+
+    if (voterInfo.canVote) {
+      return (
+        <>
+          <Buttons onClickYes={onVoteYes} onClickNo={onVoteNo} />
```

Comment:
Let's match the props here for clarity, I think `onVoteYes` is the better option